### DISCUSSION TITLE
Webstorm 2022.1 TailwindCSS 3.0.24 autocomplete does not work for utilities like top, left, bottom, right ....

### DISCUSSION THREAD

**Discussion by DanielOberlechner** [tailwindlabs/tailwindcss#8227] [Help] [answered]
Hi guys,
I have a big problem with Webstorm and TailwindCSS.
First I had a lot of problems to get autocomplete working, after some tweaking I got it to work partially.
![Bildschirmfoto 2022-04-27 um 00 41 18](https://user-images.githubusercontent.com/10922763/165848111-2d3646dd-6579-4d81-9954-c3b60b196559.png)
![Bildschirmfoto 2022-04-27 um 00 41 31](https://user-images.githubusercontent.com/10922763/165848122-824b0ed0-6c65-4ea7-ab12-555ddddb6c99.png)

Same thing on other Code Editor Visual Code and the TailwindCSS Plugin:
![Bildschirmfoto 2022-05-06 um 23 11 40](https://user-images.githubusercontent.com/10922763/167217136-8ad49341-8818-451e-8e5b-44560f011ad8.png)
![Bildschirmfoto 2022-05-06 um 23 11 17](https://user-images.githubusercontent.com/10922763/167217144-f1682a8c-1a32-4e99-a7dc-b514fa2a838d.png)

As you can see if I try to enter w-1/3 the autocomplete shows me it but if I try to use top-1/3 it doesn't work.
Even if I enter it manually it won't compile and doesn't work therefore.

Has anyone an idea how I can fix this problem?
This are my files with my configurations:

### package.json
```json
{
  "name": "it-solutions",
  "version": "0.1.1",
  "description": "IT-Solutions",
  "keywords": [
    "personal website, it-solutions, daniel oberlechner, business"
  ],
  "main": "gulpfile.js",
  "author": "Daniel Oberlechner",
  "license": "ISC",
  "start": "gulp watchFiles",
  "dependencies": {
    "cash-dom": "^8.1.0",
    "particles.js": "^2.0.0",
    "swiper": "^8.0.6",
    "tailwindcss": "^3.0.24"
  },
  "devDependencies": {
    "@tailwindcss/typography": "^0.5.2",
    "autoprefixer": "^10.4.0",
    "cssnano": "^5.0.10",
    "del": "6.0.0",
    "gulp": "^4.0.2",
    "gulp-autoprefixer": "^8.0.0",
    "gulp-concat": "^2.6.1",
    "gulp-htmlmin": "5.0.1",
    "gulp-postcss": "^9.0.1",
    "gulp-rename": "2.0.0",
    "gulp-sass": "^5.0.0",
    "gulp-sourcemaps": "3.0.0",
    "gulp-uglify": "^3.0.2",
    "gulp-uglify-es": "^3.0.0",
    "modern-css-reset": "^1.4.0",
    "postcss": "^8.3.11",
    "postcss-import": "^14.0.2",
    "postcss-load-config": "^3.1.0",
    "sass": "^1.43.4"
  },
  "browserslist": [
    "defaults"
  ],
  "postcss": {
    "parser": "sugarss",
    "map": false,
    "plugins": {
      "postcss-plugin": {},
      "autoprefixer": {},
      "tailwindcss": {}
    }
  }
}
```

### tailwind.confis.js
```javascript
module.exports = {
  development: true,
  mode: 'jit',
 future: {
    removeDeprecatedGapUtilities: false,
    purgeLayersByDefault: false
  },
  content: [
    './styles/**/*.js',
    './website/**/*.html',
  ],
  theme: {
    inset: {
    },
    container: {
      center: true,
    },
    extend: {},
  },
  variants: {
    extend: {},
  },
  plugins: [
      require('@tailwindcss/typography'),
  ],
}
```
Kind regards,
Daniel Oberlechner

**Comment by NovusOpus** [comment]
Sadly the WebStorm plugin for Tailwind CSS is maintained by JetBrains, The only Official Tailwind CSS Auto Complete, Syntax Highlighting, and Linting Tool is: [Tailwind CSS IntelliSense](https://github.com/tailwindlabs/tailwindcss-intellisense). But it's only available for VS Code.

In your case your best bet is to take a look at the [WebStorm Issue tracker site](https://youtrack.jetbrains.com/issues/WEB?q=tailwind). Here are some Discussions with suggestions:
[Discussion 1](https://youtrack.jetbrains.com/issue/WEB-54433/Tailwind-plugin-makes-code-completion-fail-for-the-whole-project) 
[Discussion 2](https://youtrack.jetbrains.com/issue/WEB-55758/Tailwind-plugin-does-not-work-No-completion-or-preview-on-hover)
This are recent discussions so maybe you can find a solution from the suggestions listed there.

**Comment by DanielOberlechner** [comment]
@NovusOpus 
Thank you so much for answering this problem.
I already googled TailwindCSS and Webstorm and found a lot of Issues, but Jetbrains claimed they fixed everything in one of those issues and then I thought maybe I'm too dumb to use the software properly or I've installed something wrong or it is only my fault somehow that it doesn't work.

I tried even with VSCode but there I have the same problem. Most classes work, but if I want to use top, left, right, bottom there is no autocomplete. And if I enter it manually it doesn't compile the classes for it. Have you a guess if this is a tailwindcss or Webstorm problem, or am I too dumb to work properly with it?

Maybe I could try to change the nodeJS version, that could do something I guess.

Kind regards and thanks for your help,
Daniel Oberlechner :)

**Comment by DanielOberlechner** [comment, marked as answer]
Found the issue you can read about it right here:
https://github.com/tailwindlabs/tailwindcss/issues/8290

Daniel
